packages/trader/src/Modules/Trading/Containers/purchase.jsx

```diff
@@ -93,24 +109,20 @@ const Purchase = observer(({ is_market_closed }) => {
             }
         } else if (vanilla_trade_type === type) {
             components.push(purchase_fieldset);
+        } else if (is_accumulator && has_open_accu_contract) {
```

Comment:
@kate-deriv I got a suggestion: let's move the 1st condition to the end like this, so that we just use `else` for the rest of the cases:
```
        if (is_vanilla && vanilla_trade_type === type) {
            components.push(purchase_fieldset);
        } else if (is_accumulator && has_open_accu_contract) {
            components.push(
                <AccumulatorsSellButton
                    is_disabled={!is_valid_to_sell}
                    onClick={onClickSellButton}
                    contract_info={active_accu_contract?.contract_info}
                    is_sell_requested={active_accu_contract?.is_sell_requested}
                    current_stake={indicative}
                    currency={currency}
                />
            );
        } else {
            switch (getContractTypePosition(type)) {
                case 'top':
                    components.unshift(purchase_fieldset);
                    break;
                case 'bottom':
                    components.push(purchase_fieldset);
                    break;
                default:
                    components.push(purchase_fieldset);
                    break;
            }
        }
```
will it work?

Reply:
Good point, let me try

Reply:
No, it didn't work because in that case we rendered 2 btns for Vanillas: we got in the first `if` and in the last `else`

Reply:
thank you for checking❤️